flatten-array.json

```diff
@@ -0,0 +1,19 @@
+{
+   "cases": [
+      {
+         "description": "flattens array with null and integers present",
+         "input": [0, 2, [[2, 3], 8, 100, null, [[null]]], -2],
+         "expected": [0, 2, 2, 3, 8, 100, -2]
```

Comment:
Here we have:

- an array
- a nested array
- a nested array inside the nested array
- nulls

I think we should build up the tests so that they are progressively more complex.